Sharing and interactive visualization chapter

kevinrue opened the pull request:
A first attempt to kick off things.

robertamezquita commented:
First PR! Thanks @AlexNghiem !

kevinrue committed:
First addition to the interactive viz chapter.

LTLA commented:
Formatting comments:

- What's up with the triple dashes? Single dashes aren't good enough for you?
- Keep an eye on the complexity of your sentences. Some of them could be broken up.
- Try not to use package names as section headings, instead describe what it does for people, e.g., "Exploring SummarizedExperiment objects".
- I'd prefer to not stick `images/iSEE-dataflow.png` in the repo. But I'll grudgingly accept it. Note that you'll have to modify `rebuild.sh` in the `Orchestrating` repo to `cp` the `images` dir.

Also, how are you planning to show the interactiveness? We could put `iframes` in the book pointing to a shiny server somewhere, if only we had a Shiny server that we could use. Probably better than having static screenshots.

kevinrue commented:
* I know double and triple dashes turn into en- and em dash, though I can never remember which ones I'm supposed to use. Probably best to stick with single dash in that case.
* I'm generally aware of the complexity of sentences. That said, a helpful advice I got from my PhD supervisors is that to get started, anything is good. Cleaning up is easier when something's there in the first place.
* Good point.
* In my defense, the image was there when I picked up the chapter. Not a problem for me to remove it. I may even have left it influence me into spending/wasting time thinking what text I could write to lead to the figure.

That is the question isn't it? Part of my issue writing anything comes exactly from that point of "how are we illustrating it". We'll think of something and then I'm sure this chapter will write itself.

LTLA commented:
> Cleaning up is easier when something's there in the first place.

What?! You mean it's not perfect to start with? 🇳🇿 👎 

> We'll think of something and then I'm sure this chapter will write itself.

I reckon we should go with `iframe`s, plus the commands required to generate them. Anything else would be crappy. Now we just have to find a sucker with a Shiny server who can host this for us.

kevinrue committed:
merge updated master branch

Merge branch 'master' into kra-interactive

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

kevinrue committed:
merge master

Merge branch 'master' into kra-interactive

# Conflicts:
#	analysis/interactive.Rmd

kevinrue committed:
avoid package name in title

kevinrue committed:
extend the interactive chapter (credits to Rob for contributions and inspiration)

LTLA commented on analysis/interactive.Rmd:
```diff
@@ -17,12 +17,82 @@ source("workflows/extractor.R")
+
+To get started, iSEE apps each provide a guided tour to walk the user through the functionality of the interface.
+
+For analysts generating the `sce` object, we recommend saving the `sce` via one of the following methods :
```
This part about data in/out seems a bit irrelevant. Better suited to the `import-export.Rmd` section.

kevinrue committed:
Merge branch 'master' into kra-interactive

kevinrue committed:
add points about interactive data viz as part of the analysis

kevinrue committed:
Merge branch 'master' of github.com:Bioconductor/OSCABase into kra-interactive

kevinrue committed:
Updated interactive chapter

federicomarini commented on analysis/interactive.Rmd:
```diff
@@ -4,90 +4,259 @@ output:
+For example, a common view for initial inspection of single-cell RNA-sequencing data sets generated by microfluidics platforms plots the library size of each cell, in decreasing order.
+An elbow in this plot generally reveals the transition between good quality cells and low quality cells or empty droplets.
+
+Another common view is to overlay the library size (in log space) as a color gradient over the result of a dimensionality reduction method (e.g., t-SNE, UMAP, ivis).
```
Do we want to add refs to the three dim red methods?

kevinrue replied:
Done

federicomarini commented on analysis/interactive.Rmd:
```diff
@@ -4,90 +4,259 @@ output:
+With `iSEE`, such interactive exploration can be guided by small guided tours - step-by-step walkthroughs of the different panels with pointers to facilitate their interpretation, from the authors' point of view.
+At any time, the viewer is still free to leave the provided tour and explore the data from their own perspective.
+All that is needed to add a tour to an `iSEE` instance is a data frame with two columns; the first column containing the UI element to be pointed to in each step of the tour, and the second one containing the text to display at that step.
+This data frame must then be provided to the `iSEE()` function via the `tour` argument. Examples of tours can be found in https://github.com/ltla/iSEE2018.
```
Should be `LTLA` all caps - although it does not make a difference in term of URL

kevinrue replied:
Done (d1814524)

federicomarini commented on analysis/interactive.Rmd:
```diff
@@ -4,90 +4,259 @@ output:
+
+With `iSEE`, this can be achieved at any time by clicking on the "Display panel settings" button under the "Diagnostics" tab in the top right corner, copying the displayed code into an R script, and sharing this chunk of code with collaborators.
+However, note that they will also need to have access to a copy of the original `sce` object used for the analysis.
+In that setup, executing the code in the script and launching iSEE with the following command will open an instance mimicking what was shown on the screen when the "Display panel settings" button was clicked.
```
`iSEE` to replace iSEE

kevinrue replied:
I caught this one and others leftover ones (d1814524)

federicomarini commented on analysis/interactive.Rmd:
```diff
@@ -4,90 +4,259 @@ output:
+
+### Annotation and identification of cell populations
+
+Recent efforts such as the Human Cell Atlas (HCA) and the NIH Human BioMolecular Atlas Program (HuBMAP) have initiated the creation of reference maps of cell types and states in the human body, in health and disease.
```
We could/should add the websites for the HCA and HuBMAP?

kevinrue replied:
Done (d1814524)

kevinrue committed:
address review comment https://github.com/Bioconductor/OSCABase/pull/1/files/3dee2b4617de69f5a56675318d83fc4496766c69#r312733869

kevinrue committed:
example for #2

kevinrue committed:
add references for interactive chapter

kevinrue committed:
Minor fixes following Fede's review.

kevinrue committed:
a few more fixes following Fede's review

LTLA commented on analysis/interactive.Rmd:
````diff
@@ -4,21 +4,265 @@ output:
+It is important to note that all `iSEE` applications remain interactive irrespective of their initial configuration; only the initial layout and state of the application is modified from the default setup. 
+
+```{r, eval=FALSE}
+iSEE(sce, redDimArgs = redDimPlotArgs, colDataArgs = colDataPlotArgs,
````
This is pretty confusing if `redDimPlotArgs` and friends are never actually created.

kevinrue replied:
Code chunk removed in 147c21c

LTLA commented on analysis/interactive.Rmd:
````diff
@@ -4,21 +4,265 @@ output:
+
+In the example below, we demonstrate that an `iSEE` app can be preconfigured to immediately display the views described above.
+
+```{r, eval=FALSE}
````
I suggest we isolate the set up from the `runApp()`, and only have the latter in an `eval=FALSE` block. This ensures that we verify that the majority of the set-up code is runnable.

kevinrue replied:
I've split the relevant chunks in 147c21c

LTLA commented on analysis/interactive.Rmd:
```diff
@@ -4,21 +4,265 @@ output:
+sce.pbmc$cluster <- factor(clust)
+
+# Identify positive markers for each cluster
+markers.pbmc.up <- findMarkers(sce.pbmc, sce.pbmc$cluster, direction="up", log.p=TRUE)
```
This now has a `sorted=FALSE` option.

kevinrue replied:
Code chunk updated in 147c21c

kevinrue committed:
integrate Aaron's review and other minor fixes.

LTLA commented on ref.bib:
```diff
@@ -1058,3 +1058,50 @@ @Article{butler2018integrating
    Month="06"
 }
 
+@article{vanDerMaaten2008tsne,
```
This is redundant, we have one of these already surely.

LTLA commented on analysis/interactive.Rmd:
```diff
@@ -4,21 +4,256 @@ output:
+## Additional resources
+
+- Publication (F1000Research, 2018): https://f1000research.com/articles/7-741/v1
+- Source code: https://github.com/csoneson/iSEE
```
We don't need these two things above. Just talk about the deployed examples.

kevinrue replied:
Done in e1f154de

LTLA replied:
You might want to expand a bit more on what you mean by deployed instances. Just have a few sentences outside of a list, and then just list out the ones we have.

kevinrue committed:
more feedback

kevinrue committed:
Merge branch 'master' into kra-interactive

kevinrue committed:
Remove duplicated reference; add missing referenced example; minor cleanup
